dotnet/src/Functions/Functions.OpenApi/RestApiOperationRunner.cs

```diff
@@ -562,5 +564,26 @@ private Uri BuildsOperationUrl(RestApiOperation operation, IDictionary<string, o
         return content;
     }
 
+    /// <summary>
+    /// Retrieves the headers from the specified HTTP response message, including content headers.
+    /// </summary>
+    /// <param name="responseMessage">The HTTP response message from which to extract headers.</param>
+    /// <returns>
+    /// A dictionary containing the headers of the HTTP response message, including content headers.
+    /// The dictionary keys are header names, and the values are collections of header values.
+    /// </returns>
+    private static Dictionary<string, IEnumerable<string>>? GetResponseHeaders(HttpResponseMessage responseMessage)
```

Comment:
May consider a `ReadOnlyDictionary` or the original `HttpResponseHeaders` if it is accessible?

```suggestion
    private static ReadOnlyDictionary<string, IEnumerable<string>>? GetResponseHeaders(HttpResponseMessage responseMessage)
```

Reply:
Agreed, is there a reason for this dictionary to be mutable?

Reply:
Having it mutable will allow users to add new headers or remove existing ones if needed.